pool/pool.go

```diff
@@ -223,12 +228,57 @@ func (p *Pool) StoreTx(ctx context.Context, tx types.Transaction, ip string, isW
 		}
 	}
 
+	gasPrices, err := p.GetGasPrices(ctx)
+	if err != nil {
+		log.Error("failed to load gas prices")
+	}
+	err = p.ValidateEffectiveGasPrice(ctx, tx, preExecutionResponse.txResponse.GasUsed, gasPrices)
+	if err != nil {
+		return err
+	}
+
 	poolTx := NewTransaction(tx, ip, isWIP)
 	poolTx.ZKCounters = preExecutionResponse.usedZkCounters
 
 	return p.storage.AddTx(ctx, *poolTx)
 }
 
+// ValidateEffectiveGasPrice validates the effective gas price
+func (p *Pool) ValidateEffectiveGasPrice(ctx context.Context, tx types.Transaction, preExecutionGasUsed uint64, gasPrices GasPrices) error {
+	breakEvenGasPrice, err := p.effectiveGasPrice.CalculateBreakEvenGasPrice(tx.Data(), tx.GasPrice(), preExecutionGasUsed, gasPrices.L1GasPrice)
+	if err != nil {
+		log.Errorf("error calculating break even gas price: %v", err)
+		return err
+	}
+	margin := big.NewInt(0).SetUint64(p.cfg.EffectiveGasPrice.MarginBreakEven)
+	breakEvenGasPricePercentApplied := big.NewInt(0).Mul(breakEvenGasPrice, margin)
+	breakEvenGasPricePercentApplied = breakEvenGasPricePercentApplied.Div(breakEvenGasPricePercentApplied, big.NewInt(percentFactor))
+	log.Infof("EGP breakEvenGasPrice: tx:%s breakEvenGasPrice:%v margin:%d breakEvenGasPricePercentApplied:%v L1GasPrice:%d L2GasPrice:%d tx.GasPrice():%v EGPenabled:%t",
```

Comment:
In the sequencer side I start this log with "egp-log: " ("egp-log: breakEvenGasPrice: .....") maybe we can use the same format to be consistent, anyway we can review is "egp-log" is ok or use other prefix